sympy/core/sympify.py

```diff
@@ -359,15 +363,18 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     if isinstance(a, CantSympify):
         raise SympifyError(a)
     cls = getattr(a, "__class__", None)
-    if cls is None:
-        cls = type(a)  # Probably an old-style class
-    conv = converter.get(cls, None)
-    if conv is not None:
-        return conv(a)
 
+    #check for user defined converters
+    for superclass in getmro(cls):
+        try:
+            return _external_converter[superclass](a)
```

Comment:
It's much more likely that we will find what we are looking for in `_sympy_converter` so I would swap the order of these two loops.

Reply:
I was thinking that if the user wants to override SymPy's conversion that should be possible (it is currently).

Reply:
I'm considering to put the `_sympy_converter` and `_external_converter` inside the same loop (but still `_external_converter` first)

Reply:
> I'm considering to put the `_sympy_converter` and `_external_converter` inside the same loop (but still `_external_converter` first)

Currently whichever entry from the mro first matches will be chosen so I guess checking all of the external ones before any of the internal ones could lead to a different result e.g. if `_sympy_converter` matches lower in the mro than `_external_converter`. Maybe that's impossible though...

Reply:
Yes, I think so.

Reply:
We should probably have some tests that none of this has potentially altered the behaviour.

Reply:
I'll open a different PR for that, since we need to make sure that those tests pass in the current master too.